File: backend/src/dto/response/cabinet.per.section.response.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { CabinetInfoResponseDto } from './cabinet.info.response.dto';

/**
 * 특정 층에 존재하는 사물함 정보
 */
export class CabinetsPerSectionResponseDto {
  @ApiProperty({
    description: '사물함의 섹션 종류 (오아시스 등)',
    example: 'Oasis',
  })
  section: string; // 사물함의 섹션 종류 (오아시스 등)

  @ApiProperty({
    description: '해당 섹션에 존재하는 사물함들',
    type: [CabinetInfoResponseDto],
  })
  cabinets: CabinetInfoResponseDto[]; // 해당 섹션에 존재하는 사물함들
}

File: backend/src/main/java/org/ftclub/cabinet/dto/response/CabinetPerSectionResponseDto.java
package org.ftclub.cabinet.dto.response;

import lombok.*

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CabinetsPerSectionResponseDto {
    /**
     * FIXME:
     * 위치 정보 수정
     */
    private String section;
    private CabinetInfoResponseDto[] cabinets;
}